[Sharing 2.0] di fixes

rullzer opened the pull request:
Since we also want to use the share manager in the not to distant future from the cli we should only inject generic containers and construct all the other stuff from there.

Lets see what ci says so far.

For https://github.com/owncloud/core/issues/19331

rullzer committed:
[Sharing 2.0] Removed unused DI stuff

The share manager etc should not care about filtering stuff. They should
return what is asked for them.

rullzer committed:
[Sharing 2.0] Default share provider only generic DI

No injection of userfolders etc. Only generic DI components
(IRootFolder) etc should be used to make sure we can also run this from
the cli

rullzer commented:
Review time. @schiesbn @DeepDiver1975 @nickvergessen @PVince81

rullzer commented:
Ah this means of course that we need to do more permission checking in the OCS endpoint. Let me add that first.

nickvergessen commented on apps/files_sharing/api/ocssharewrapper.php:
```diff
@@ -26,25 +26,22 @@ class OCSShareWrapper {
-		               \OC::$server->getUserFolder(),
-		               \OC::$server->getURLGenerator());
+		return new Share20OCS(
+			new \OC\Share20\Manager(
```
this should be in the `\OC::$server` as well I think, but maybe you want to wait until ... :question:

rullzer replied:
Basically until it is a bit more stable ;)

nickvergessen commented on lib/private/share20/defaultshareprovider.php:
```diff
@@ -218,14 +218,14 @@ private function createShare($data) {
+		$owner = $path->getOwner();
+		$share->setShareOwner($owner);
+
+		$path = $this->rootFolder->getUserFolder($owner->getUID())->getById((int)$data['file_source'])[0];
```
The array could also be empty... maybe you need to handle that case separately?
Also the TODO comment above is then resolved?

rullzer replied:
yes... I need to think about that.. basically if the array is empty there is a data inconsitency... and maybe just throw an exception that explictly says that.

nickvergessen commented on lib/private/share20/manager.php:
```diff
@@ -40,37 +36,19 @@ class Manager {
-		$this->currentUser = $user;
-		$this->userManager = $userManager;
-		$this->groupManager = $groupManager;
+	public function __construct(
```
missing doc

nickvergessen commented on lib/private/share20/manager.php:
```diff
@@ -246,7 +218,7 @@ public function getSharedWithMe($shareType = null, $page=0, $perPage=50) {
 	 * @return Share
 	 *
-	 * @throws ShareNotFoundException
+	 * @throws ShareNotFound
```
Also throws `\Exception` as seen below, either fix the docs, or adjust the body

rullzer committed:
[Sharing 2.0] Move authentication to the OCS API

rullzer committed:
[Sharing 2.0] Fix phpdoc etc

rullzer committed:
[Sharing 2.0] Use the rootfolder to get the path of a share

We need to use the rootfolder here since we also other people than the
shareOwner can request a share.

rullzer commented:
* Fixed some phpdoc stuff
* Moved authentication over to OCS endpoint. So manager can really care about managing stuff
* Rebased

And CI is stuff happy :)

@DeepDiver1975 should help for your CI stuff.

DeepDiver1975 commented:
:+1:

schiesbn commented:
looks good :+1:

rullzer commented:
@DeepDiver1975 please merge
